ltiauthenticator/lti13/handlers.py

```diff
@@ -223,17 +223,23 @@ def post(self):
 
         client_id = args["client_id"]
         self.log.debug(f"client_id is {client_id}")
-        redirect_uri = guess_callback_uri(
-            "https", self.request.host, self.hub.server.base_url
+
+        redirect_uri = "{proto}://{host}{path}".format(
+            proto=self.request.protocol,
```

Comment:
Should the protocol ever be something else than `https://`? I understand that this is where the user will be redirected, and we shouldn't redirect users to send data during an authentication phase to a http destination right?

If this server received the request after it was passed on from a proxy that terminated TLS (transformed https:// to http://), then I figure the end user should still be redirected to `https://`.

I figured this may have been a fix to ensure the test suite passed, but I tried using `"https"` in a fork of mine, and tests still passed.

What do you think?

Reply:
`http` may be used in a local test deployment to avoid the need to set up certificates, which is why I made the change.

Since the logic is the same as for [`OAuthentticator.get_callback_url`](https://github.com/jupyterhub/oauthenticator/blob/5e3007841f23a8415e0d9cc6e11af9de056a5793/oauthenticator/oauth2.py#L453) I would expect it to work also behind a reverse proxy terminating TLS. I haven't tested it though.

Reply:
Just checked the tornado docs: If the reverse proxy correctly sets the `X-Scheme` header, then `request.protocol` should resolve to `https` if that is used on the public side of the proxy.

Reply:
Wow excellent followup! This then LGTM!

Do you consider this ready to be merged?